tests/test_api.py

```diff
@@ -0,0 +1,386 @@
+):
+    client = app_with_admin_user.test_client()
+    setup_mocks(config, monkeypatch)
+
+    # Try to update the resource
+    lookup = config["json"].get(config.get("lookup_field") or "id")
+    response = client.patch(f"{resource}/{lookup}", json=config.get("PATCH_json"))
+    if "PATCH" in config["allowed_methods"]:
+        # Need to match etag
+        assert response.status_code == 428
+        response = client.patch(
+            f"{resource}/{lookup}",
+            json=config.get("PATCH_json"),
+            headers={"if-match": ETAG},
+        )
+        assert response.status_code == 200
+        # Check that the record was updated
+        item = db_with_records.query(config["model"]).one().__dict__
+        assert_dict_contains(item, config["PATCH_json"])
+    else:
+        assert response.status_code in (404, 405)
+
+
+@pytest.mark.parametrize("resource, config", resource_requests.items())
+def test_item_put(resource, config, app_with_admin_user, db_with_records, monkeypatch):
+    setup_mocks(config, monkeypatch)
+    client = app_with_admin_user.test_client()
+
+    # Try to PUT the resource - this is disallowed for all resources.
+    lookup = config["json"].get(config.get("lookup_field") or "id")
+    response = client.put(f"{resource}/{lookup}", json=config["json"])
+    if "PUT" in config["allowed_methods"]:
+        assert response.status_code == 200
+        assert response.json == config["json"]
+    else:
+        assert response.status_code in (404, 405)
+
+
+@pytest.mark.parametrize("resource, config", resource_requests.items())
+def test_item_delete(
+    resource, config, app_with_admin_user, db_with_records, monkeypatch
+):
+    setup_mocks(config, monkeypatch)
+    client = app_with_admin_user.test_client()
+
+    # Try to DELETE the resource - this is disallowed for all resources.
+    lookup = config["json"].get(config.get("lookup_field") or "id")
+    response = client.delete(f"{resource}/{lookup}", headers={"if-match": ETAG})
+    if "DELETE" in config["allowed_methods"]:
+        assert response.status_code == 204
+    else:
+        assert response.status_code in (404, 405)
+
+
+@pytest.mark.parametrize("resource, config", resource_requests.items())
+def test_resource_filters(
+    resource, config, app_with_admin_user, db_with_records, monkeypatch
+):
+    if "filters" not in config:
+        return
+
+    setup_mocks(config, monkeypatch)
+    client = app_with_admin_user.test_client()
+
+    one_response = client.get(resource, query_string=config["filters"]["one"])
+    assert one_response.status_code == 200
+    assert len(one_response.json["_items"]) == 1
+    item = one_response.json["_items"][0]
+    assert_dict_contains(item, config["json"])
+
+    empty_response = client.get(resource, query_string=config["filters"]["empty"])
+    assert empty_response.status_code == 200
+    assert empty_response.json["_items"] == []
+
+
+@pytest.mark.parametrize("resource, config", resource_requests.items())
```

Comment:
maybe move this filtering 
```
if "additional_records" not in config:
        return
```
into parametrize call so no dummy tests will be run
```suggestion
@pytest.mark.parametrize("resource, config", [r,c for r,c in resource_requests.items() if if "additional_records" in c])
```

Reply:
Oh, cool feature! Will do.